source/common/quic/http_datagram_handler.cc

```diff
@@ -0,0 +1,86 @@
+#include "source/common/quic/http_datagram_handler.h"
```

Comment:
I _think_ we're going to want to support building out datagram support separately from HTTP/3 support unless @RyanTheOptimist weighs in and says it's not worth it.

We've already got a knob for HTTP/3 support being compiled out (see http3 =False) and here's an example PR where I add a knob for compiling stats out which is a combination of having a bazel select for library files, having the bazel select set a copt we can use for #defines and having #defines in appropriate places.  I'd be inclined to turn it off by default in Envoy mobile builds by setting that new option true by default in mobile/.bazelrc

https://github.com/envoyproxy/envoy/pull/26086/files

Reply:
Thanks for sharing the relevant PR. I will wait until Ryan shares his thoughts on this.

Reply:
I agree that adding a knob to not build Capsule, independent of the HTTP/3 know seems like the right choice, specifically for Envoy Mobile.

Reply:
I added a commit to allow Envoy Mobile to compile out datagram support. Let me know if there are any issues.